org.eclipse.jdt.core/model/org/eclipse/jdt/core/compiler/CompilerConfiguration.java

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+* Copyright (c) 2024 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.core.compiler;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.internal.compiler.impl.CompilerOptions;
+
+/**
+ * @since 3.38
+ */
+public class CompilerConfiguration {
```

Comment:
What about this? Are we exposing this to the outside world too? If not, we need @noreference here as well. Otherwise, this could use some documentation.

Reply:
This is not only used by internal AbstractImageBuilder. It will be also used by clients for retrieving the compiler options.

Reply:
But aren't all the expected clients basically implementing AbstractImageBuilder (so already consuming this internal package) ?
While I think all this will need to be API at some point, as it's early stage, I think we'd rather keep as much as possible in "internal" packages, until we feel comfortable enough with the design to turn it API.

Reply:
The clients don't implement AbstractImageBuilder, but the ECJ internal Compiler class. Yes, we can put the new CompilerConfiguration within internal package since it remains accessible from other plugin.

Just curious what's guideline to split the public package and internal package in the JDT codebase?

Reply:
Ah yes, my bad!
The public package become API that we can then hardly modify and that we have to maintain as such. internal packages are basically not supported externally and can be changed more freely. So in general, unless there is a good reason for something to become API, we try to keep things internal as it makes further maintenance significantly easier.
But here, it can be a case where a new API class makes sense. I think it's actually good to have an actual object formalizing portable standard compiler configuration that could be used in multiple clients, rather than passing some options and (potentially unknown) participants.

Reply:
Exactly. Currently, project compiler-related settings are scattered across multiple places. Introducing a centralized class to formalize them would make them easily portable for other clients to reuse.